### ISSUE TITLE
The last captcha in results is not right 

### ISSUE THREAD

**Issue by jnyuzs** [Dobiasd/frugally-deep#40]
Hi,
    Please help me! Thanks!
    A model file and a image file in attach zip file .
[captcha.zip](https://github.com/Dobiasd/frugally-deep/files/1959485/captcha.zip)
   The model is for captch,one input ,four outputs.
    use python,the result is krc8,it is right,  
    but use c++,the result is krc0, the forth captcha is zero,it is not right.

     the follow is c++ code:
     	cv::Mat image = cv::imread(fileName); //改成你自己的图片路径
	cv::resize(image, image, cv::Size(80, 40));
	cv::cvtColor(image, image, cv::COLOR_BGR2RGB);
	std::cout << cv::Size(80, 40) << std::endl;
	assert(image.isContinuous());
	const auto input = fdeep::tensor3_from_bytes(image.ptr(),
		image.rows, image.cols, image.channels(), 0.0f, 1.0f);
	const auto result = model.predict({ input });

 the follow is python code:
    data = np.empty((1,H, W, 3), dtype="float32")
    raw_img = preprocess.load_img(fileName)
    img=cv2.resize(raw_img,(80,40))
    data[0,:, :, :] = img/255.0
    result= model.predict(data)

**Comment by Dobiasd** [comment]
I downloaded your model and converted it as follows:
```bash
python3 ./keras_export/convert_model.py keras_model7.h5 keras_model7.json
```

Then I used your code (slightly adjusted to make it a valid minimal example):
```cpp
#include "fdeep/fdeep.hpp"
#include <opencv2/opencv.hpp>
int main()
{
    const std::string fileName = "krc8.jpg";
    const std::string modelPath = "keras_model7.json";
    cv::Mat image = cv::imread(fileName);
    cv::resize(image, image, cv::Size(80, 40));
    cv::cvtColor(image, image, cv::COLOR_BGR2RGB);
    std::cout << cv::Size(80, 40) << std::endl;
    assert(image.isContinuous());
    const auto input = fdeep::tensor3_from_bytes(image.ptr(),
        static_cast<std::size_t>(image.rows),
        static_cast<std::size_t>(image.cols),
        static_cast<std::size_t>(image.channels()),
        0.0f, 1.0f);
    const auto model = fdeep::load_model(modelPath);
    const auto result = model.predict({ input });
    for (const auto& res : result)
    {
        std::cout << fdeep::internal::tensor3_max_pos(res).z_ << std::endl;
    }
}
```

And got this output:
```bash
[80 x 40]
Parsing json ... done. elapsed time: 0.029485 s
Running test 1 of 1 ... done. elapsed time: 0.006105 s
20
27
12
0
```

Then I ran the adjusted python code:
```python
import numpy as np
import cv2
from keras.models import load_model
from keras import backend as K
from keras.preprocessing.image import load_img
H = 40
W = 80
fileName = "krc8.jpg"
modelPath = "keras_model7.h5"
model = load_model(modelPath)
data = np.empty((1, H, W, 3), dtype="float32")
raw_img = np.asarray(load_img(fileName))
img = cv2.resize(raw_img,(80,40))
data[0,:, :, :] = img/255.0
result = model.predict(data)
print(np.argmax(result, axis=2))
```

and received this result:
```
[[20]
 [27]
 [12]
 [ 8]]
```

So ideed, the last of the four outputs is different. This should not be the case. I will investigate the issue and get back to you with the results.

**Comment by Dobiasd** [comment]
The problem is the following:

The sum (i.e. divisor) in the `softmax` layer is `0` with simply float precision.

If you replace
```cpp
#include "fdeep/fdeep.hpp"
```

with
```cpp
#define FDEEP_FLOAT_TYPE double
#include "fdeep/fdeep.hpp"
```

in your C++ code, you receive the correct output:

```
[80 x 40]
Parsing json ... done. elapsed time: 0.026906 s
Running test 1 of 1 ... done. elapsed time: 0.010137 s
20
27
12
8
```
